portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,20 +37,30 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    int numComm = Integer.parseInt(request.getParameter("num-choice"));
+    
+    if (numComm > 20) {
+      response.setContentType("text/html");
+      response.getWriter().println("You cannot see over 20 comments");
+      return;
+    }
```

Comment:
Nice touch!  Keep in mind though, your JS code is expecting a json object as the response.  If you respond with html you're going to hit errors in your promise callback.  A better pattern here is to return some JSON object like: {error: "... Your message ..."}.  The JS code can then check if there's an error before processing the response, and display it if necessary

Reply:
I liked this feature, why'd you remove it?

Reply:
i wasn't able to figure out how to get it to print right and i dint want to get stuck on it for too long.